File: Pset2 – 1 Readability/readability.c
#import <stdio.h>
#import <cs50.h>
#import <string.h>
#import <math.h>

int main(void)
{

    // Request text

    string text = get_string("Text: ");

    // Get total length of string

    int length = strlen(text);

    // Initialize my 3 core variables

    int letters = 0;
    int words = 1;
    int sentences = 0;

    for (int a = 0; a < length; a++)
    {

        // Count letters

        if (((text[a] >= 48 && text[a] <= 48) || (text[a] >= 65 && text[a] <= 90) || (text[a] >= 97 && text[a] <= 122)))
        {
            letters++;
        }

        // Count words

        else if (text[a] == 32)
        {
            words++;
        }

        // Count sentences

        else if ((text[a] == 46) || (text[a] == 33) || ((text[a] == 63) && (text[a] != 32)))
        {
            sentences++;
        }

    }

    // My averages

    float L = ((float) letters / (float) words) * 100;
    float S = ((float) sentences / (float) words) * 100;

    // Formula

    int index = round(0.0588 * L - 0.296 * S - 15.8);

    // Grades

    if (index < 1)
    {
        printf("Before Grade 1\n");
    }

    else if (index >= 16)
    {
        printf("Grade 16+\n");
    }

    else if (index >= 1 && index < 16)
    {
        printf("Grade %i\n", index);
    }

}

File: Pset6 – 4 Readability/readability.py
from cs50 import get_string

# Get a string from the user
text = get_string("Text: ")

# The length of the string
length = len(text)

# The three core values
letters = 0
words = 1
sentences = 0

for a in range(length):

    # Calculate the letters
    if (((ord(text[a]) >= 48 and ord(text[a]) <= 48)
        or (ord(text[a]) >= 65 and ord(text[a]) <= 90)
        or (ord(text[a]) >= 97 and ord(text[a]) <= 122))):
        letters += 1

    # Calculate the words
    elif ord(text[a]) == 32:
        words += 1

    # Calculate the sentences
    elif ((ord(text[a]) == 46) or (ord(text[a]) == 33)
        or ((ord(text[a]) == 63) and (ord(text[a]) != 32))):
        sentences += 1

# Calculate the averages
L = (float(letters) / float(words)) * 100
S = (float(sentences) / float(words)) * 100

# The formula itself
index = round(0.0588 * L - 0.296 * S - 15.8)

# The results
if index < 1:
    print("Before grade 1")

elif index >= 16:
    print("Grade 16+")

elif index >= 1 and index < 16:
    print(f"Grade {index}")
